packages/flutter/lib/src/material/expansion_panel.dart

```diff
@@ -395,16 +406,29 @@ class _ExpansionPanelListState extends State<ExpansionPanelList> {
     if (widget._allowOnlyOnePanelOpen) {
       final ExpansionPanelRadio pressedChild = widget.children[index];
 
+      // Determine if an ExpansionPanelRadio was opened prior, invoking
```

Comment:
Suggested rewording: If another ExpansionPanelRadio was already open, apply its expansionCallback (if any) to false, because it's closing.

However, it's possible that all of the expansion panels were open because previously this was an ordinary expansion panel list?

Reply:
If I understand the question correctly, I think you're asking about how this logic works with the case where ExpansionPanelList --> press --> ExpansionPanelList.radio

I believe `_handlePressed` only opens/closes the radio panel in question, and additionally closing a previous panel if it was an `ExpansionPanelList.radio`. Lets say tapping a panel of ExpansionPanelList with all panels open sets state that updates the widget to behave like ExpansionPanelList.radio:
1) First, `_handlePressed` will be called and trigger the callback once.
2) Then, `didUpdateWidget` would update the widget and set the open radio panel, if any

Thus, `expansionCallback` will not be called for any of the prior open `ExpansionPanelList` panels. Should this be a case we should consider?

Reply:
That makes sense; in this (corner) case there's no need to call the expansionCallback on the formerly open panels, since they were part of an ordinary ExpansionPanelList and its callback wouldn't be called in this case.